### PR TITLE
[ADD] module pos_require_customer

### PR DESCRIPTION
**Pull request by JosDeGraeve**
This module allows you to require a customer for each pos order.  In the
pos configuration, you can choose to require the customer for pos
orders.

If a customer is not selected, the pos ui will display an error message.
In the backend the customer field is required when needed.

### CODE REVIEW

File: pos_require_customer/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+This module allows you to require a customer for each pos order.  In the
+pos session configuration, you can choose to require the customer for pos
+orders.
+
+If a customer is not selected, the pos ui will display an error message.
+In the backend the customer field is required when needed.
+
+    """,
+    'author': 'Apertoso NV',
```
Comment (legalsylvain) [inline pos_require_customer/__openerp__.py]: Could you add OCA as coauthor ?
Sample : https://github.com/OCA/pos/blob/8.0/hw_customer_display/__openerp__.py#L78

**Comment by legalsylvain** [comment]
Hi. thanks for sharing this module. 
Code review : OK for me. 
Just on point, could you add a constraint _constraint, on pos.order based on session_id, partner_id, to create the same check if the user create pos order by back end GUI ?
 Regards.

**Comment by legalsylvain** [comment]
>  In the backend the customer field is required when needed.

In fact I don't see this feature implemented.

**Comment by JosDeGraeve** [comment]
Hi,

Tnx for the review.  The advertised functionality is present now :)

I actually implemented it on a version running with a client, but merged an older version.  Sorry for the confusion.

I 'll further try to add some tests.

File: pos_require_customer/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+##############################################################################
+
+{
+    'name': 'Point of Sale Require Customer',
+    'version': '1.0.1',
+    'category': 'Point Of Sale',
+    'sequence': 6,
+    'summary': 'Point of Sale Require Customer',
+    'description': """
+Require customer for pos
```
Comment (pedrobaeza) [inline pos_require_customer/__openerp__.py:28]: Please extract this to a README.rst file and use OCA's template: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: pos_require_customer/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from . import pos
```
Comment (pedrobaeza) [inline pos_require_customer/__init__.py]: Put encoding and move the models file to a folder `models`

**Comment by pedrobaeza** [comment]
I wasn't able to try on runbot. I got a too much recursion error.

@legalsylvain, do you know about this error?

**Comment by JosDeGraeve** [comment]
Imo, i got i all right now.

**Comment by pedrobaeza** [comment]
@JosDeGraeve, it still continues with the same error starting the POS:

seleccion_001

**Comment by legalsylvain** [comment]
Hi, @JosDeGraeve. The problem @pedrobaeza mentions seems to be out of the scope of your PR. The branch V7 / all seems to be broken in Runbot.
It is urgent to add test on web client readable by Travis / Runbot.

**Comment by legalsylvain** [comment]
@JosDeGraeve : could you rebase or merge the last upstream, please. It could fix @pedrobaeza remarks I guess.

**Comment by legalsylvain** [comment]
:+1: Code review and test. Thanks a lot !

**Comment by pedrobaeza** [comment]
Can you please name it _pos_customer_mandatory_ for avoiding verbs in the module name?

**Comment by StefanRijnhart** [comment]
:+1:. @pedrobaeza can we be lenient about the name? It has been a long review cycle already.

**Comment by pedrobaeza** [comment]
That last change is very easy. I can do it at merge time if the author can't do it, but I prefer to keep the best possible names now that the module is not yet on the repo.

**Comment by StefanRijnhart** [comment]
OK, but then I think that pos_customer_required is more appropriate given the prevalence of the 'required' terminology over 'mandatory' in Odoo.

**Comment by pedrobaeza** [comment]
OK, I agree with the name. Should I proceed with the merge?

**Comment by legalsylvain** [comment]
Ow ! 
I guess I'll have to re-rename my V9 PR ... ;-) (https://github.com/OCA/pos/pull/74)

**Comment by StefanRijnhart** [comment]
@pedrobaeza that is fine by me. Make sure you search+replace all occurrences of pos_require_customer in the module (but I don't really have to tell you that, do I).

**Comment by pedrobaeza** [comment]
Renamed and merged!

**Pull request merged**